### PR TITLE
Invite User or Institution

### PR DESCRIPTION
**Pull request by devduarte**
Fixes: #1356 

-Create frontend flow to allow the user to choose to invite an institution or a user.

## Before this PR
Screen Shot 2021-12-16 at 09 28 10

## After this PR
Screen Shot 2021-12-16 at 09 28 35

Screen Shot 2021-12-16 at 09 28 52

Screen Shot 2021-12-16 at 09 29 06

Screen Shot 2021-12-16 at 09 29 28

### CODE REVIEW

**Comment by diegoliberman** [comment]
@devduarte thank you so much for the screenshots in the PR! :)

**Comment by devduarte** [comment]
On this last commit I hid the new invitations flow just in case, for some reason, we need to merge it to main. To try the new feature replace this line located in `app/views/users/index.haml`

` = react_component "AddUserLink", institution_types: Institution.kinds, roles: @available_roles_hash, context: @navigation_context.to_hash`

for 

`= react_component "AddUserOrInstitutionLink", institution_types: Institution.kinds, roles: @available_roles_hash, context: @navigation_context.to_hash`

**Review by ysbaddaden — COMMENTED** [review]
React, you're drunk. That's an awful lot of JavaScript just to reimplement native `<form>` :sob: 

There could be a feature-flip to enable the feature. For example a simple ENV var such as `FEATURE_INVITES=1` and an internal check:

```ruby
react_component ENV["FEATURE_INVITES"] == "1" ? "AddUserOrInstitutionLink" : "AddUserLink",
  institution_types: Institution.kinds,
  roles: @available_roles_hash,
  context: @navigation_context.to_hash
```

And add `FEATURE_INVITES=1` to `docker-dev.env` and to Rancher in staging. When we're ready to deploy the feature in production, we can either add the variable or remove it from the code then deploy.

**Comment by devduarte** [comment]
with the last commit, the modal title changes accordingly:

Screen Shot 2021-12-16 at 11 51 16

Screen Shot 2021-12-16 at 11 52 54

Screen Shot 2021-12-16 at 11 53 14

**Comment by ftarulla** [comment]
> with the last commit, the modal title changes accordingly:
> 
> Screen Shot 2021-12-16 at 11 51 16
> 
> Screen Shot 2021-12-16 at 11 52 54
> 
> Screen Shot 2021-12-16 at 11 53 14

I think the last screen should reflect the institution type selected in the first view

File: app/assets/javascripts/components/institution_invite_form.js.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      type: newValue
+    });
+  },
+
+  back: function() {
+    this.props.modalPresenterStep()
+  },
+
+  next: function() {
+    this.props.manufacturerStep()
```
Comment (ftarulla) [inline app/assets/javascripts/components/institution_invite_form.js.jsx:20]: The name `manufacturerStep` should reflect the selected institution type

File: app/assets/javascripts/components/user_invite_form.js.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    if(this.state.includeMessage)
+      data.message = this.state.message;
+    $.ajax({
+      url: '/users',
+      method: 'POST',
+      data: data,
+      success: function () {
+        this.closeModal();
+        window.location.reload(true); // reload page to update users table
```
Comment (ftarulla) [inline app/assets/javascripts/components/user_invite_form.js.jsx:29]: Do we lose the `state` when reloading? 🤔

**Review by ftarulla — APPROVED** [review]

**Pull request merged**